src/api.ts

```diff
@@ -210,6 +220,10 @@ export interface MongoDBOIDCPlugin {
    * for automatic token refreshing.
    */
   destroy(): Promise<void>;
+
+  on: MongoDBOIDCPluginEventEmitter['on'];
+
+  off?: MongoDBOIDCPluginEventEmitter['off'];
```

Comment:
This seems to duplicate the the functionality of `logger`, no?

Reply:
But logger is for logging, I don't think it's good to mix those concepts

Reply:
Then let’s update the name of that property or the docs for it, `.logger` is absolutely intended to be used for programmatic information about state updates (and it’s used that way in Compass and devtools-connect, too).

Reply:
Also seems like `mongodb-oidc-plugin:state-updated` doesn't provide state information, the only public interface in plugin available returns a serialized version of it, while we need the actual tokens to be able to pass them to the requests. Errors are also not part of the oidc-plugin state, so this will require us to subscribe to more logger events instead of having an explicit event interface to subscribe to that is not connected to unrelated functionality

Reply:
Right – would just doing a token request after `mongodb-oidc-plugin:state-updated` work to get the actual tokens?

I’d really like to avoid breaking up the encapsulation here and having “wires stick out” of the plugin instance more than necessary

Reply:
FWIW I don't think providing an explicit event interface is "wires stick out" and calling a refresh method on an arbitrary logger event is not, but I don't have a strong enough preference if that's what we intend to have as a plugin interface

Reply:
Yeah, maybe the naming isn’t ideal here, but really, `.logger` is meant for programmatic access. If it wasn’t, it wouldn’t have a fully typed and documented API. :)